pkg/cluster/cluster.go

```diff
@@ -450,6 +457,16 @@ func UpdateCluster(client *cmv1.ClustersClient, clusterID string, config Spec) e
 		}
 		clusterBuilder = clusterBuilder.Proxy(clusterProxyBuilder)
 	}
+	if config.ClusterWideProxy.NoProxy != nil {
+		if (config.ClusterWideProxy.HTTPProxy != nil && *config.ClusterWideProxy.HTTPProxy != "") ||
+			(config.ClusterWideProxy.HTTPSProxy != nil && *config.ClusterWideProxy.HTTPSProxy != "") ||
+			cluster.Body().Proxy().HTTPProxy() != "" || cluster.Body().Proxy().HTTPSProxy() != "" {
```

Comment:
I wonder if we should skip this check here and leave it to the backend to validate there is a proxy when setting noproxy, that way we don't have to fetch the cluster first and can unify adding the NoProxy section with the previous condition

Reply:
Most of the code that I was involved with in both in ocm-cli and rosa, contains as much validation as possible to avoid sending the request to CS and let it decide instead of the client.

Reply:
but to make this validation work we need to send a request to CS to get the cluster first, so we aren't saving a request here - in fact we're adding a request to all update calls, even if we don't touch the proxy